crates/swift-bridge-ir/src/codegen/codegen_tests/string_codegen_tests.rs

```diff
@@ -262,3 +262,50 @@ func __swift_bridge__some_function () -> UnsafeMutableRawPointer {
         .test();
     }
 }
+
+/// Test code generation for Swift function that takes and returns an owned String argument.
+mod extern_swift_func_takes_and_returns_string {
+    use super::*;
+
+    fn bridge_module_tokens() -> TokenStream {
+        quote! {
+            mod foo {
+                extern "Swift" {
+                    fn some_function (value: String) -> String;
+                }
+            }
+        }
+    }
+
+    fn expected_rust_tokens() -> ExpectedRustTokens {
+        ExpectedRustTokens::Contains(quote! {
+            pub fn some_function (value: String) -> String {
+                unsafe {
+                    Box :: from_raw (unsafe { __swift_bridge__some_function (swift_bridge :: string :: RustString (value) . box_into_raw ()) }) . 0
+                }
+            }
+        })
+    }
+
+    const EXPECTED_SWIFT_CODE: ExpectedSwiftCode = ExpectedSwiftCode::ContainsAfterTrim(
+        r#"
+@_cdecl("__swift_bridge__$some_function")
+func __swift_bridge__some_function (_ value: UnsafeMutableRawPointer) -> UnsafeMutableRawPointer {
+    { let rustString = some_function(value: RustString(ptr: value)).intoRustString(); rustString.isOwned = false; return rustString.ptr }()
```

Comment:
Why do we need to call `intoRustString` if, according to your PR body, `some_function` would return a `RustString` here?

Reply:
@NiwakaDev waiting on this question then after that we can merge this and solve the test stuff in a separate PR

Reply:
Oh and also please add an example bridge module to your PR body so that it's more clear that the bridge module uses String and not RustString

Reply:
Any availability to look at this? One this is addressed I can merge

Reply:
I am working on it, but I'm slightly busy this month. Sorry...

Reply:
> Why do we need to call intoRustString if, according to your PR body, some_function would return a RustString here?

Nevermind, the reason is that we support returning anything that can be converted into a RustString.